src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -1179,6 +1189,14 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                                 if (tombstone != null && ChangeVectorUtils.GetConflictStatus(item.ChangeVector, tombstone.ChangeVector) == ConflictStatus.AlreadyMerged)
                                     continue;
 
+                                // tombstone structure means doc-id | REC-SEP | ...
+                                var keySpan = attachmentTombstone.Key.AsReadOnlySpan();
+                                int endOfDocumentId = keySpan.IndexOf(SpecialChars.RecordSeparator);
+                                Debug.Assert(endOfDocumentId != -1);
+                                string documentId = Encoding.UTF8.GetString(keySpan[..endOfDocumentId]);// will throw if no separator found 
```

Comment:
sth that worries me is that we have a piece of code like that here, don't we have some sort of a helper class that centralizes this? Can't we have some AttachmentKeyHelper or sth like that?

Reply:
Done